src/App.tsx

```diff
@@ -8,32 +10,46 @@ function getRandomName(): string {
 }
 
 export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
+  const [clockName, setTimerName] = useState('Clock-0');
+  const [hasClock, setClockOn] = useState(true);
+  const [prevClockName, setPrevName] = useState('Clock-0');
+
+  let timerId = 0;
+
+  const detectKeyDown = (e: MouseEvent) => {
+    e.preventDefault();
+
+    if (e.button === 2) {
+      setClockOn(false);
+    } else {
+      setClockOn(true);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('click', detectKeyDown);
+    window.addEventListener('contextmenu', detectKeyDown);
+  });
```

Comment:
```suggestion
  useEffect(() => {
    window.addEventListener('click', detectKeyDown);
    window.addEventListener('contextmenu', detectKeyDown);
  }, []);
```
Add empty dependencies array if you want for useEffect to act like a componentDidMount.
For now, it creates a new eventListener after each change, which is not correct. Also, don't forget to remove all eventListeners and intervals before component unmount